### ISSUE TITLE
Ascendex - Orders are stuck on cancelling when using multiple orders and low refresh time in Pure Market Making.

### ISSUE THREAD

**Issue by rapcmia** [CoinAlpha/hummingbot#3777]
**Describe the bug**
When using Pure market-making with multiple orders and low order refresh time, there are instances that orders are stuck at canceling for a while. This happened as well even when there are hanging orders enabled.

**Steps To Reproduce**
1. Create a pure market-making strategy with multiple orders
2. Set order refresh time to 10s (or 15s) and enable hanging orders (optional)
3. Start the bot and wait when it tries to cancel the order 

On this recording, It was more than 2mins before it created orders again:
https://user-images.githubusercontent.com/73840223/126601683-c03d7203-f4c9-46b9-af67-fec54f6e4d98.mp4

```
2021-07-22 14:23:56,215 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Created LIMIT_MAKER BUY order HMBot-B-DOGE-USDT-1626935036004515 for 59 DOGE-USDT.
2021-07-22 14:23:56,215 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935036.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "DOGE-USDT", "amount": "59", "price": "0.1898410", "order_id": "HMBot-B-DOGE-USDT-1626935036004515", "exchange_order_id": "HMBot17acce2886eU112449668204515", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "ascend_ex"}

# Order cancel event triggered
2021-07-22 14:24:05,985 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003512. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,985 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003513. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,985 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003514. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,985 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003515. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,985 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003516. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,986 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004515. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,986 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004516. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:24:05,986 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004517. [clock=2021-07-22 06:24:06+00:00]
2021-07-22 14:25:07,012 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003512. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,012 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003513. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003514. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003515. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036003516. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004515. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004516. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:25:07,013 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Cancelling the limit order HMBot-B-DOGE-USDT-1626935036004517. [clock=2021-07-22 06:25:07+00:00]
2021-07-22 14:26:00,293 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036003512.
2021-07-22 14:26:00,306 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036003512", "exchange_order_id": "HMBot17acce28864U112449668203512", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,306 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036003513.
2021-07-22 14:26:00,324 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036003513", "exchange_order_id": "HMBot17acce28866U112449668203513", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,324 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036003514.
2021-07-22 14:26:00,341 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036003514", "exchange_order_id": "HMBot17acce28868U112449668203514", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,341 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036003515.
2021-07-22 14:26:00,360 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036003515", "exchange_order_id": "HMBot17acce2886aU112449668203515", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,360 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036003516.
2021-07-22 14:26:00,377 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036003516", "exchange_order_id": "HMBot17acce2886cU112449668203516", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,378 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036004515.
2021-07-22 14:26:00,395 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036004515", "exchange_order_id": "HMBot17acce2886eU112449668204515", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,395 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036004516.
2021-07-22 14:26:00,412 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036004516", "exchange_order_id": "HMBot17acce28870U112449668204516", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 14:26:00,412 - 8084 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-DOGE-USDT-1626935036004517.
2021-07-22 14:26:00,430 - 8084 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935160.0, "order_id": "HMBot-B-DOGE-USDT-1626935036004517", "exchange_order_id": "HMBot17acce28872U112449668204517", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}

# Order created again
2021-07-22 14:26:01,010 - 8084 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Creating 8 bid orders at (Size, Price): ['59 DOGE, 0.1905041 USDT', '59 DOGE, 0.190485 USDT', '59 DOGE, 0.1904659 USDT', '59 DOGE, 0.1904469 USDT', '59 DOGE, 0.1904278 USDT', '59 DOGE, 0.1904088 USDT', '59 DOGE, 0.1903897 USDT', '59 DOGE, 0.1903706 USDT']
```

On this recording, it's the same behavior again but this time I have hanging orders enabled
https://user-images.githubusercontent.com/73840223/126602165-af02922f-9e80-4c32-849c-c6fc8773548e.mp4

Even when the hanging orders are filled, the bot still trying to cancel the order:
![image](https://user-images.githubusercontent.com/73840223/126602287-0a466f29-90aa-4e51-a5b1-16151be707a9.png)

**Release version**
dev-0.42.0

**Attachments**
[pmm-test.zip](https://github.com/CoinAlpha/hummingbot/files/6860545/pmm-test.zip)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by rxlxrxsx** [comment]
Got the same issue as well. 

<img width="1422" alt="Screen Shot 2021-07-22 at 2 29 12 PM" src="https://user-images.githubusercontent.com/64377055/126604514-ed19f9cb-4d1b-411d-9abb-08dc3002daf0.png">

In this screenshot you'll see that the orders are successfully cancelled on time in the exchange.

<img width="1117" alt="Screen Shot 2021-07-22 at 2 30 31 PM" src="https://user-images.githubusercontent.com/64377055/126604521-694c662c-40ce-4341-ac50-0653bd962a34.png">

From the log file you can see here that it took 2 minutes before it was successfully cancelled in the client.

```python
2021-07-22 06:28:02,000 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001844. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001905. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001934. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001955. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001977. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-S-CLOUT-USDT-1626935272002155. [clock=2021-07-22 06:28:02+00:00]
2021-07-22 06:28:02,417 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-CLOUT-USDT-1626935272001934.
2021-07-22 06:28:02,427 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935282.0, "order_id": "HMBot-B-CLOUT-USDT-1626935272001934", "exchange_order_id": "HMBot17acce62246U871539022301934", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:29:03,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001844. [clock=2021-07-22 06:29:03+00:00]
2021-07-22 06:29:03,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001905. [clock=2021-07-22 06:29:03+00:00]
2021-07-22 06:29:03,002 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001955. [clock=2021-07-22 06:29:03+00:00]
2021-07-22 06:29:03,002 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-B-CLOUT-USDT-1626935272001977. [clock=2021-07-22 06:29:03+00:00]
2021-07-22 06:29:03,002 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Cancelling the limit order HMBot-S-CLOUT-USDT-1626935272002155. [clock=2021-07-22 06:29:03+00:00]

# 2 minute gap

2021-07-22 06:30:00,161 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-CLOUT-USDT-1626935272001844.
2021-07-22 06:30:00,170 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935400.0, "order_id": "HMBot-B-CLOUT-USDT-1626935272001844", "exchange_order_id": "HMBot17acce62242U871539022301844", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:30:00,171 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-CLOUT-USDT-1626935272001905.
2021-07-22 06:30:00,179 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935400.0, "order_id": "HMBot-B-CLOUT-USDT-1626935272001905", "exchange_order_id": "HMBot17acce62244U871539022301905", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:30:00,180 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-CLOUT-USDT-1626935272001955.
2021-07-22 06:30:00,188 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935400.0, "order_id": "HMBot-B-CLOUT-USDT-1626935272001955", "exchange_order_id": "HMBot17acce62248U871539022301955", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:30:00,189 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-B-CLOUT-USDT-1626935272001977.
2021-07-22 06:30:00,197 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935400.0, "order_id": "HMBot-B-CLOUT-USDT-1626935272001977", "exchange_order_id": "HMBot17acce6224aU871539022301977", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:30:00,198 - 14805 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Successfully cancelled order HMBot-S-CLOUT-USDT-1626935272002155.
2021-07-22 06:30:00,206 - 14805 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626935400.0, "order_id": "HMBot-S-CLOUT-USDT-1626935272002155", "exchange_order_id": "HMBot17acce6224cU871539022302155", "event_name": "OrderCancelledEvent", "event_source": "ascend_ex"}
2021-07-22 06:30:01,001 - 14805 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (CLOUT-USDT) Creating 5 bid orders at (Size, Price): ['0.2 CLOUT, 100.08 USDT', '0.2 CLOUT, 99.98 USDT', '0.2 CLOUT, 99.88 USDT', '0.2 CLOUT, 99.78 USDT', '0.2 CLOUT, 99.68 USDT']
```

Logs and config: [ascendex_log_conf.zip](https://github.com/CoinAlpha/hummingbot/files/6860663/ascendex_log_conf.zip)

**Comment by pulkitsharma** [comment]
If anybody is facing issue reproducing, you can also try 1s order_refresh_time and run two PMM bots with cumulative value more than available balance. Eg 100USDT each bot and total balance should be less than 200usdt.

**Comment by dennisocana** [comment]
Fixed in version 0.42 release
